app/response_processor.py

```diff
@@ -46,105 +52,129 @@ def service_name(self, url=None):
+            raise QuarantinableError
 
         elif decrypted_json.get("survey_id") == "census":
             self.logger.info("About to publish receipt into ctp queue")
-            queue_ok = self.ctp_publisher.publish_message(dumps(receipt_json), secret=settings.SDX_COLLECT_SECRET)
+
+            try:
+                self.ctp_publisher.publish_message(dumps(receipt_json),
+                                                   secret=settings.SDX_COLLECT_SECRET)
+            except PublishMessageError as e:
+                self.logger.error("Unsuccesful publish", error=e)
+                raise RetryableError
+
         else:
-            self.logger.info("About to publish receipt into rrm queue")
-            queue_ok = self.rrm_publisher.publish_message(dumps(receipt_json), secret=settings.SDX_COLLECT_SECRET)
 
-        if not queue_ok:
-            raise RetryableError()
+            try:
+                self.logger.info("About to publish receipt into rrm queue")
+                self.rrm_publisher.publish_message(dumps(receipt_json),
+                                                   secret=settings.SDX_COLLECT_SECRET)
+            except PublishMessageError as e:
+                self.logger.error("Unsuccesful publish", error=e)
+                raise RetryableError
 
         self.logger.info("Receipt published")
 
     def decrypt_survey(self, encrypted_survey):
         self.logger.info("Decrypting survey")
-        response = self.remote_call(settings.SDX_DECRYPT_URL, data=encrypted_survey)
+        response = self.remote_call(settings.SDX_DECRYPT_URL,
+                                    data=encrypted_survey)
         try:
             self.response_ok(response)
-        except BadMessageError:
-            # Translate the Bad Message into a Decrypt Error to force quarantine
-            raise DecryptError
+        except ClientError:
+            self.logger.error("Survey decryption unsuccessful. Quarantining Survey.")
+            raise QuarantinableError
 
         self.logger.info("Survey decryption successful")
         return response.json()
 
     def validate_survey(self, decrypted_json):
         self.logger.info("Validating survey")
-        self.response_ok(self.remote_call(settings.SDX_VALIDATE_URL, json=decrypted_json))
+        self.response_ok(self.remote_call(settings.SDX_VALIDATE_URL,
+                                          json=decrypted_json))
         self.logger.info("Survey validation successful")
 
     def store_survey(self, decrypted_json):
         self.logger.info("Storing survey")
-        self.response_ok(self.remote_call(settings.SDX_RESPONSES_URL, json=decrypted_json))
+        response = self.remote_call(settings.SDX_RESPONSES_URL,
+                                    json=decrypted_json)
+        try:
+            self.response_ok(response)
+        except ClientError:
+            self.logger.error("Survey storage unsuccessful. Quarantining Survey.")
+            raise QuarantinableError
```

Comment:
It was throwing a BadMessageError if store gives back 400 response. I don't think it should be retrying once sdx-store gives a 400 response and neither should it be throwing it away because it has gone through decrypt and validate.